tests/drivers/uart/uart_emul/prj.conf

```diff
@@ -1,3 +1,5 @@
 CONFIG_ZTEST=y
+CONFIG_EVENTS=y
 CONFIG_SERIAL=y
 CONFIG_UART_INTERRUPT_DRIVEN=y
+CONFIG_UART_ASYNC_API=y
```

Comment:
It would be better to create another test in testcase.yaml so both async and non-async APIs are tested.

Reply:
It just enables the async API. As far as i understand it, the APIs can coexist. Otherwise the test cases would fail, as each test case checks one of the APIs. Or am i missing something?

Reply:
I still think it is a good idea to test without enabling the API at all.

Reply:
I did split this into three test configurations